conda/gateways/connection/session.py

```diff
@@ -42,6 +46,33 @@ def close(self):
         raise NotImplementedError()
 
 
+@lru_cache(None)
+def get_channel_name_from_url(url: str) -> str | None:
```

Comment:
I notice this is different than https://github.com/conda/conda/blob/main/conda/models/channel.py#L34 . Am unfamiliar with MultiChannel.

Reply:
Probably want to use this instead:

https://github.com/conda/conda/blob/bdacdbb6079cc877e52a322d511c666aa8e89276/conda/models/channel.py#L451